### PR TITLE
libimobiledevice: update all packages

### LINKED ISSUE #342728: Update request: idevicerestore 1.0.0+date=2023-05-23 → 1.0.0-unstable-2024-07-09
- Package name: idevicerestore
- Latest released version:
https://github.com/libimobiledevice/idevicerestore/releases
https://github.com/libimobiledevice/idevicerestore/commits/master/

- Current version on the unstable channel: 1.0.0+date=2023-05-23
- Current version on the stable/release channel: [1.0.0+date=2023-05-23](https://search.nixos.org/packages?channel=24.05&type=packages&query=idevicerestore)

- [x] Checked the [nixpkgs pull requests](https://github.com/NixOS/nixpkgs/pulls)

**Notify maintainers**
@nh2 

-----

Note for maintainers: Please tag this issue in your PR.

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### LINKED ISSUE #342729: Package request: libtatsu 1.0.3
**Project description**

Library handling the communication with Apple's Tatsu Signing Server (TSS).

**Additional information**

Needed for https://github.com/NixOS/nixpkgs/issues/342728

```
checking for libusbmuxd-2.0 >= 2.0.2... yes
checking for libplist-2.0 >= 2.6.0... yes
checking for libimobiledevice-glue-1.0 >= 1.3.0... yes
checking for libtatsu-1.0 >= 1.0.3... no
configure: error: Package requirements (libtatsu-1.0 >= 1.0.3) were not met:

No package 'libtatsu-1.0' found

Consider adjusting the PKG_CONFIG_PATH environment variable if you
installed software in a non-standard prefix.
```

**Metadata**

* homepage URL: https://github.com/libimobiledevice/libtatsu
* source URL: https://github.com/libimobiledevice/libtatsu
* license:  LGPL-2.1 license 
* platforms: unix

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### LINKED ISSUE #342731: Update request: usbmuxd 1.1.1+date=2023-05-05 → 1.1.1-unstable-2024-09-15
- Package name: usbmuxd
- Latest released version:
https://github.com/libimobiledevice/usbmuxd/releases
https://github.com/libimobiledevice/usbmuxd/commits/master/

- Current version on the unstable channel: 1.1.1+date=2023-05-05
- Current version on the stable/release channel: [1.1.1+date=2023-05-05](https://search.nixos.org/packages?channel=24.05&show=usbmuxd&type=packages&query=usbmuxd)

- [x] Checked the [nixpkgs pull requests](https://github.com/NixOS/nixpkgs/pulls)

**Notify maintainers**

-----

Note for maintainers: Please tag this issue in your PR.

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### LINKED ISSUE #342736: Update request: ideviceinstaller 1.1.1+date=2023-04-30 → 1.1.1-unstable-2024-05-18
- Package name: ideviceinstaller
- Latest released version:
https://github.com/libimobiledevice/ideviceinstaller/releases
https://github.com/libimobiledevice/ideviceinstaller/commits/master/

- Current version on the unstable channel: 1.1.1+date=2023-04-30
- Current version on the stable/release channel: [1.1.1+date=2023-04-30](https://search.nixos.org/packages?channel=24.05&type=packages&query=+ideviceinstaller)

- [x] Checked the [nixpkgs pull requests](https://github.com/NixOS/nixpkgs/pulls)

**Notify maintainers**
@aristid

-----

Note for maintainers: Please tag this issue in your PR.

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### LINKED ISSUE #342738: Update request:  ifuse 1.1.4+date=2022-04-04 → 1.1.4-unstable-2023-01-06
- Package name: ifuse
- Latest released version:
https://github.com/libimobiledevice/ifuse/releases
https://github.com/libimobiledevice/ifuse/commits/master/

- Current version on the unstable channel: 1.1.4+date=2022-04-04
- Current version on the stable/release channel: [1.1.4+date=2022-04-04](https://search.nixos.org/packages?channel=24.05&type=packages&query=ifuse)

- [x] Checked the [nixpkgs pull requests](https://github.com/NixOS/nixpkgs/pulls)

This request is more about the rename of the package conforming to https://github.com/NixOS/nixpkgs/pull/234201 than anything else, since the latest commit isn't any meaningful upgrade.

**Notify maintainers**

-----

Note for maintainers: Please tag this issue in your PR.

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by Frontear**
## Description of changes

Closes #342738
Closes #342736
Closes #342731
Closes #342728
Closes #342729

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/tools/misc/usbmuxd/default.nix [hunk 1]
```diff
@@ -54,4 +55,4 @@ stdenv.mkDerivation rec {
     maintainers = [ ];
```
Comment (dotlambda) [inline pkgs/tools/misc/usbmuxd/default.nix:55]: Wanna add yourself?

**Review by dotlambda — CHANGES_REQUESTED** [review]
Make sure the commit messages follow https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#commit-conventions

File: pkgs/tools/misc/usbmuxd/default.nix (hunk 1 above)
Comment (Frontear) [reply pkgs/tools/misc/usbmuxd/default.nix:55]: yep i'll add myself

**Comment by Frontear** [comment]
> Make sure the commit messages follow [`master`/pkgs/README.md#commit-conventions](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md?rgh-link-date=2024-09-19T00%3A30%3A16Z#commit-conventions)

Most of the version strings are too long to fit in the title (or is the limitation a neovim highlighting thing?)

**Comment by Frontear** [comment]
How did commits from other authors get in here? Did I mess up my rebasing?

**Comment by Frontear** [comment]
> Make sure the commit messages follow [`master`/pkgs/README.md#commit-conventions](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md?rgh-link-date=2024-09-19T00%3A30%3A16Z#commit-conventions)

Fixed!

**Comment by Frontear** [comment]
Messed up rebasing I believe, apologies! I'll re-open the PR shortly

**Pull request closed without merging**